### PR TITLE
AESinkAudioTrack: Try to aim 200 ms buffer and 50 ms periods

### PR DESCRIPTION
**Pull request by fritsch**
Adjust Audio-Buffers to match other platforms.

A sink works as follows:
It has a buffer, which consists of certain amount of periods of audio.
If period size is too large, e.g. only 2 fit into a buffer, we might underrun if just 1 period comes too late.
If period size is far too small, you get a cpu problem, with e.g. 10 ms period you have to come back 100 times per second.

On pulse and on Alsa we aim at 200 to 400 ms buffer and have periods of around 50 ms. This has proven as a good ratio among cpu load, underruns, etc.

For Android, we used two times the minimum buffer allowed and half of it as period. The PR at hand increases the buffer size and tries to keep the period size as it was before to not create more load but to be more failsafe, when e.g. listening to high quality audio and switching towards the menu where a thread could get stuck for too long.

Old Audiotrack: 64 ms buffer and 32 ms period
New Audiotrack (on same hardware): 250 ms buffer and therefore 60 ms periods.

This solves issues for @HitcherUK when listening to music and going into the menus.

### CODE REVIEW

**Comment by fritsch** [comment]
In an ideal world, it would be good if the periods / buffers actually reflect the hw settings, but those are - in opposite to ALSA - not exposed for audio track.

**Comment by HitcherUK** [comment]
Backport for 18.5 as this is a fix please.

**Comment by fritsch** [comment]
Needs a lot regression testing on several devices.

For example: AMLogic devices feedback (the old ones), FireTV 4K, etc.

Also: TrueHD, DTS-HD, etc. needs testing. Those are quite sensitiv when it comes to buffer / periods ratio.

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp [hunk 1]
```diff
@@ -461,17 +461,21 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
     }
     else
     {
-      m_min_buffer_size *= 2;
-
       m_format.m_frameSize = m_format.m_channelLayout.Count() * (CAEUtil::DataFormatToBits(m_format.m_dataFormat) / 8);
       m_sink_frameSize = m_format.m_frameSize;
-      m_format.m_frames = (int)(m_min_buffer_size / m_format.m_frameSize) / 2;
+      // aim at 200 ms buffer but at max 300 ms (2 * 0.15s) and 50 ms periods
+      m_audiotrackbuffer_sec = (double)(m_min_buffer_size) / (m_sink_frameSize * m_sink_sampleRate);
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp]: use C++ style cast

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp [hunk 2]
```diff
@@ -461,17 +461,21 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
...
+      while (m_audiotrackbuffer_sec < 0.15)
+      {
+        m_min_buffer_size *= 2;
+        m_audiotrackbuffer_sec = (double)(m_min_buffer_size) / (m_sink_frameSize * m_sink_sampleRate);
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp]: use C++ style cast

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp [hunk 3]
```diff
@@ -461,17 +461,21 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
...
+      }
+      // division by 4 -> 4 periods into one buffer
+      m_format.m_frames = (int)(m_min_buffer_size / m_format.m_frameSize) / 4;
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp]: use C++ style cast

**Comment by HitcherUK** [comment]
@fritsch DTS-MA and Dolby-HD both working correctly.

**Comment by CiNcH83** [comment]
Might this also have an impact on the audio resample behavior for live sources? Frequency/rr fluctuations are still high on BRAVIA, causing audible distortion.

**Comment by fritsch** [comment]
@CiNcH83 Don't think so - rather sounds like a shitty mediacodec implementation which fluctuates all over the place.

**Review by peak3d — APPROVED** [review]
Tested on wetek HUB with no issues.

**Pull request merged**